### PR TITLE
Simplify @polkadot/keyring KeyringPair

### PR DESCRIPTION
**Pull request by axelchalon**
Modify the KeyringPair interface to be consistent with [changes](https://github.com/polkadot-js/ui/pull/144) made to `@polkadot/ui-keyring`

### CODE REVIEW

File: packages/keyring/src/pair/index.ts [hunk 1]
```diff
@@ -65,8 +65,7 @@ const verify = (type: KeypairType, message: Uint8Array, signature: Uint8Array, p
 export default function createPair (type: KeypairType, { publicKey, secretKey }: PairInfo, meta: KeyringPair$Meta = {}, encoded: Uint8Array | null = null): KeyringPair {
   return {
     type,
-    address: (): string =>
-      encodeAddress(publicKey),
+    address: encodeAddress(publicKey),
```
Comment (jacogr) [inline packages/keyring/src/pair/index.ts]: There is an issue with this - if the prefix changes, this is not taken into account for the address. SWo basically, depending on where the set is called by the user, if may be incorrect.

File: packages/keyring/src/pair/index.ts [hunk 2]
```diff
@@ -82,15 +81,14 @@ export default function createPair (type: KeypairType, { publicKey, secretKey }:
...
+    get meta () {
+      return meta;
+    },
+    isLocked: (!secretKey || secretKey.length === 0),
     lock: (): void => {
       secretKey = new Uint8Array(0);
     },
-    publicKey: (): Uint8Array =>
-      publicKey,
+    publicKey,
```
Comment (jacogr) [inline packages/keyring/src/pair/index.ts]: The same as above applies to the publicKey - this is set as we decode (so you could instantiate with an empty value, and unlock/decode to extract).

File: packages/keyring/src/pair/index.ts (hunk 1 above)
Comment (jacogr) [reply packages/keyring/src/pair/index.ts]: Think I am actually wrong here - passed by ref, so changes should reflect. Would probably be good adding an explicit test around this then. 

1.  Get address, compare to known 42 prefix 
2. Set prefix to 68
3. Compare to known 68 prefix

File: packages/keyring/src/pair/index.ts (hunk 2 above)
Comment (jacogr) [reply packages/keyring/src/pair/index.ts]: Probably also incorrect...

File: packages/keyring/src/pair/index.ts (hunk 1 above)
Comment (jacogr) [reply packages/keyring/src/pair/index.ts]: So this is the test that (is not available, but would worin in master, but doesn't after this -

```js
// in pair/index.spec.ts

// once master is merged this import would be @polkadot/util-crypto
import { setAddressPrefix } from '../address';

describe('pair', () => {
  ...
  it('allows encoding of address with different prefixes', () => {
    expect(keyring.alice.address).toEqual(
      '5GoKvZWG5ZPYL1WUovuHW3zJBWBP5eT8CbqjdRY4Q6iMaQua'
    );

    setAddressPrefix(68);

    expect(keyring.alice.address).toEqual(
      '7sGUeMak588SPY2YMmmuKUuLz7u2WQpf74F9dCFtSLB2td9d'
    );
  });
```

File: packages/keyring/src/pair/index.ts (hunk 2 above)
Comment (axelchalon) [reply packages/keyring/src/pair/index.ts]: I think this is still an issue; verifying.

Comment (jacogr) [reply packages/keyring/src/pair/index.ts]: Yes, also wanted to check that one - probably best just making it a getter anyway, that way we know. (As per my comments, when not completely awake, I do sometimes miss the plot as to when things are by value vs by ref, at least a getter won't be surprising as to intent)

**Pull request merged**